python-sdk/nuscenes/scripts/export_kitti.py

```diff
@@ -0,0 +1,347 @@
+# nuScenes dev-kit.
+# Code written by Holger Caesar, 2019.
+# Licensed under the Creative Commons [see licence.txt]
+
+"""
+This script converts nuScenes data to KITTI format and KITTI results to nuScenes.
+It is used for compatibility with software that uses KITTI-style annotations.
+
+We do not encourage this, as:
+- KITTI has only front-facing cameras, whereas nuScenes has a 360 degree horizontal fov.
+- KITTI has no radar data.
+- The nuScenes database format is more modular.
+- KITTI fields like occluded and truncated cannot be exactly reproduced from nuScenes data.
+- KITTI has different categories.
+
+Limitations:
+- We don't specify the KITTI imu_to_velo_kitti projection in this code base.
+- We map nuScenes categories to nuScenes detection categories, rather than KITTI categories.
+- Attributes are not part of KITTI and therefore set to '' in the nuScenes result format.
+- Velocities are not part of KITTI and therefore set to 0 in the nuScenes result format.
+- This script uses the `train` and `val` splits of nuScenes, whereas standard KITTI has `training` and `testing` splits.
+
+This script includes three main functions:
+- nuscenes_gt_to_kitti(): Converts nuScenes GT annotations to KITTI format.
+- render_kitti(): Render the annotations of the (generated or real) KITTI dataset.
+- kitti_res_to_nuscenes(): Converts a KITTI detection result to the nuScenes detection results format.
+
+To launch these scripts run:
+- python export_kitti.py nuscenes_gt_to_kitti --nusc_kitti_dir ~/nusc_kitti
+- python export_kitti.py render_kitti --nusc_kitti_dir ~/nusc_kitti
+- python export_kitti.py kitti_res_to_nuscenes --nusc_kitti_dir ~/nusc_kitti
+
+To work with the original KITTI dataset, use these parameters:
+ --nusc_kitti_dir /data/sets/kitti --split training
+
+See https://www.nuscenes.org/object-detection for more information on the nuScenes result format.
+"""
+import os
+import json
+from typing import List, Dict, Any
+
+from pyquaternion import Quaternion
+import numpy as np
+import fire
+from PIL import Image
+import matplotlib.pyplot as plt
+
+from nuscenes.nuscenes import NuScenes
+from nuscenes.utils.geometry_utils import transform_matrix
+from nuscenes.utils.geometry_utils import BoxVisibility
+from nuscenes.utils.data_classes import LidarPointCloud, Box
+from nuscenes.utils.splits import create_splits_logs
+from nuscenes.utils.kitti import KittiDB
+from nuscenes.eval.detection.utils import category_to_detection_name
+
+
+class KittiConverter:
+    def __init__(self,
+                 nusc_kitti_dir: str = '~/nusc_kitti',
+                 cam_name: str = 'CAM_FRONT',
+                 lidar_name: str = 'LIDAR_TOP',
+                 image_count: int = 10,
+                 nusc_version: str = 'v1.0-mini',
+                 split: str = 'mini_train'):
```

Comment:
Should we be using the non mini versions as the default arguments?

Reply:
Better to use mini so users can quickly check if this works.